meetnearme/api
# Integrate clerk auth

## Description
This PR implements authentication via Clerk using clerk-go-sdk v2.

## Changes
- Implement clerk middleware
- Implement login and signup pages
- Implement error handling response for HTTPErrors and displaying fallback htmx error components.\

## How to test this PR
- Checkout the branch
- Run `npm run dev`
- Navigate to Home page
- Click on **Account** in the navbar. (You should be redirected to the login page)
- Click on signup and sign up using Email
- Post successful signup, Login using Email from the login screen
- Click on **Account** in the navbar again. You should see your account name and userID on screen. Refer screenshots.

## Screenshots
<img width="1624" alt="Screenshot 2024-06-07 at 4 53 27 PM" src="https://github.com/meetnearme/api/assets/27922601/7fdc04b6-5d06-482a-82ae-38d3f0a41a0d">

<img width="1624" alt="Screenshot 2024-06-07 at 4 53 41 PM" src="https://github.com/meetnearme/api/assets/27922601/4f8340ec-b076-4d68-8ba4-585e59b40a18">

<img width="1624" alt="Screenshot 2024-06-07 at 4 54 13 PM" src="https://github.com/meetnearme/api/assets/27922601/1320f0bb-9300-442c-b7e0-cd27ef54d907">
